robottelo/ui/scparams.py

```diff
@@ -20,3 +20,126 @@ def _search_locator(self):
+                locators['sc_parameters.merge_overrides'],
+                matcher_merge_overrides
+            )
+        if matcher_merge_default is not None:
+            self.assign_value(
+                locators['sc_parameters.merge_default'],
+                matcher_merge_default
+            )
+        if matcher_merge_avoid is not None:
+            self.assign_value(
+                locators['sc_parameters.avoid_duplicates'],
+                matcher_merge_avoid
+            )
+        if matcher:
+            self.add_matcher(matcher)
+        if hidden_value is not None:
+            self.assign_value(
+                locators['sc_parameters.hidden_value'], hidden_value)
+        self.click(common_locators['submit'])
+
+    def add_matcher(self, matcher_list):
+        """Adding new matcher to Smart Variable
+
+        :param matcher_list: List of matchers to be added to smart class
+            parameter. Each element is a dictionary of next format:
+            {
+            'matcher_attribute': 'attr_type=attr_value',
+            'matcher_value': 'value',
+            (optional) 'matcher_puppet_default': True
+            }
+
+        """
+        for i, matcher in enumerate(matcher_list, start=1):
+            self.click(locators['sc_parameters.add_matcher'])
+            matcher_attribute = matcher['matcher_attribute'].split('=')
+            self.assign_value(
+                locators['sc_parameters.matcher_attribute_type'] % i,
+                matcher_attribute[0]
+            )
+            self.assign_value(
+                locators['sc_parameters.matcher_attribute_value'] % i,
+                matcher_attribute[1]
+            )
+            self.assign_value(
+                locators['sc_parameters.matcher_value'] % i,
+                matcher['matcher_value']
+            )
+            if 'matcher_puppet_default' in matcher.keys():
+                self.assign_value(
+                    locators['sc_parameters.matcher_puppet_default'] % i,
+                    matcher['matcher_puppet_default']
+                )
+
+    def validate_smart_class_parameter(
+            self, name, puppet_class, field_name, field_value):
+        """Checks if selected smart class parameter has necessary field values
+        on the index page
+
+        :param str name: Name of Smart Class Parameter to be validated
+        :param str field_name: Smart Class Parameter field that should be
+            validated (e.g. 'puppet_class' or 'overrides_number')
+        :param str field_value: Expected value for specified field
+        """
+        self.search_sc_params(name, puppet_class)
+        searched = self.wait_until_element(
+            locators['sc_parameters.table_value'] % field_value)
+        if searched is None:
+            raise UIError(
+                'Smart Class Parameter "{0}" field in the table has not "{1}" '
+                'value.'.format(field_name, field_value)
+            )
```

Comment:
To be honest I still like to leave code as it is. In that way I will have much better error handling rather than `got False, but wanted True` or lesser code in case you like to add custom message to each assertion
However, your thoughts sounds pretty logical. Okay